docs/samples/Microsoft.ML.Samples/Dynamic/Transforms/Text/ProduceHashedNgrams.cs

```diff
@@ -0,0 +1,69 @@
+﻿using System;
+using System.Collections.Generic;
+using Microsoft.ML.Data;
+using Microsoft.ML.Transforms.Text;
+
+namespace Microsoft.ML.Samples.Dynamic
+{
+    public static class ProduceHashedNgrams
+    {
+        public static void Example()
+        {
+            // Create a new ML context, for ML.NET operations. It can be used for exception tracking and logging, 
+            // as well as the source of randomness.
+            var mlContext = new MLContext();
+
+            // Create a small dataset as an IEnumerable.
+            var samples = new List<TextData>()
+            {
+                new TextData(){ Text = "This is an example to compute Ngrams using hashing." },
+                new TextData(){ Text = "Ngram is a sequence of 'N' consecutive words/tokens." },
+                new TextData(){ Text = "ML.NET's ProduceHashedNgrams API produces count of Ngrams and hashes it as an index into a vector of given bit length." },
+                new TextData(){ Text = "The hashing schem reduces the size of the output feature vector" },
+                new TextData(){ Text = "which is useful in case when number of Ngrams is very large." },
+            };
+
+            // Convert training data to IDataView.
+            var dataview = mlContext.Data.LoadFromEnumerable(samples);
+
+            // A pipeline for converting text into numeric hashed Ngram features.
+            // The following call to 'ProduceHashedNgrams' requires the tokenized text/string as input.
+            // This is acheived by calling 'TokenizeIntoWords' first followed by 'ProduceHashedNgrams'.
+            // Please note that the length of the output feature vector depends on the 'numberOfBits' settings.
+            var textPipeline = mlContext.Transforms.Text.TokenizeIntoWords("Tokens", "Text")
+                .Append(mlContext.Transforms.Conversion.MapValueToKey("Tokens"))
+                .Append(mlContext.Transforms.Text.ProduceHashedNgrams("NgramFeatures", "Tokens", numberOfBits: 8, ngramLength: 3, useAllLengths: false));
+            
+            // Fit to data.
+            var textTransformer = textPipeline.Fit(dataview);
+
+            // Create the prediction engine to get the features extracted from the text.
+            var predictionEngine = mlContext.Model.CreatePredictionEngine<TextData, TransformedTextData>(textTransformer);
+
+            // Convert the text into numeric features.
+            var prediction = predictionEngine.Predict(samples[0]);
+
+            // Print the length of the feature vector.
+            Console.WriteLine($"Number of Features: {prediction.NgramFeatures.Length}");
+
+            // Print the first 10 feature values.
+            Console.Write("Features: ");
+            for (int i = 0; i < 10; i++)
+                Console.Write($"{prediction.NgramFeatures[i]:F4}  ");
+
+            //  Expected output:
+            //   Number of Features: 256
```

Comment:
>//   Number of Features: 256 [](start = 12, length = 28)

if you use maximumNumberOfInverts you can also show up slot names as in just ngrams.

Reply:
I tried that but its bit complex to represent it on console as many ngram can fall into one slot. 

---
In reply to: [271903797](https://github.com/dotnet/machinelearning/pull/3177#discussion_r271903797) [](ancestors = 271903797)

Reply:
Well you can control with that parameter how many you actually want to remember, right? So if you put 1, you will have only one of them

---
In reply to: [271929298](https://github.com/dotnet/machinelearning/pull/3177#discussion_r271929298) [](ancestors = 271929298,271903797)